src/ml_tooling/baseclass.py

```diff
@@ -266,10 +268,21 @@ def make_prediction(
         try:
             if proba:
                 data = self.estimator.predict_proba(x)
-                columns = [f"Probability Class {col}" for col in range(data.shape[1])]
+                columns = [
+                    f"Probability Class {col}" for col in self.estimator.classes_
+                ]
             else:
                 data = self.estimator.predict(x)
                 columns = ["Prediction"]
+            if threshold is not None:
+                y_prob = data if proba else self.estimator.predict_proba(x)
```

Comment:
How does this code work if we use `data` - this would be a 1D array? Does it work correctly if we set threshold to 0.5? and run the threshold code against `np.array([1, 2, 1, 0])`

Reply:
Then we could go back to default 0.5 if that works correctly

Reply:
It works correctly with 0.5 threshold i think, the question is if we can use data, ill have to check that out

Reply:
This wasnt possible without transforming the 1d array into the shape of predict_proba